tools/rtpsrelay/GuidPartitionTable.cpp

```diff
@@ -94,4 +94,25 @@ GuidPartitionTable::Result GuidPartitionTable::insert(const OpenDDS::DCPS::GUID_
   return result;
 }
 
+bool GuidPartitionTable::has_any_partition(const OpenDDS::DCPS::GUID_t& guid, const StringSet& partitions) const
+{
+  ACE_GUARD_RETURN(ACE_Thread_Mutex, g, mutex_, false);
+  const auto iter = guid_to_partitions_cache_.find(guid);
+  if (iter == guid_to_partitions_cache_.end()) {
```

Comment:
I think this should be treated as a cache miss and the cache populated.  The lookup function has code for doing this.